lib/assembly-image/image.rb

```diff
@@ -131,86 +131,124 @@ def create_jp2(params = {})
+      # jp2 creation command
+      kdu_bin = 'kdu_compress'
+      options = []
+      options << '-jp2_space sRGB' if samples_per_pixel == 3
+      options += kdu_compress_default_options
+      options << "Clayers=#{layers.to_s}"
+      jp2_command = "#{kdu_bin} #{options.join(' ')} -i '#{@tmp_path}' -o '#{output}'"
+      result=`#{jp2_command} 2>&1`
+      raise "JP2 creation command failed: #{jp2_command} with result #{result}" unless $?.success?
 
-        # construct a path to the input profile, which might exist either in the gem itself or in the tmp folder
-        input_profile_file_gem = File.join(path_to_profiles,"#{input_profile}.icc")
-        input_profile_file_tmp = File.join(tmp_folder,"#{input_profile}.icc")
-        input_profile_file = File.exists?(input_profile_file_gem) ? input_profile_file_gem : input_profile_file_tmp
+      tmp_tiff_file.delete unless preserve_tmp_source
 
-        # if input profile was extracted and does not matches an existing known profile either in the gem or in the tmp folder,
-        # we'll issue an imagicmagick command to extract the profile to the tmp folder
-        unless File.exists?(input_profile_file)
-          input_profile_extraction_command = "MAGICK_TEMPORARY_PATH=#{tmp_folder} convert '#{@path}'[0] #{input_profile_file}" # extract profile from input image
-          result=`#{input_profile_extraction_command} 2>&1`
-          raise "input profile extraction command failed: #{input_profile_extraction_command} with result #{result}" unless $?.success?
-          raise 'input profile is not a known profile and could not be extracted from input file' unless File.exists?(input_profile_file) # if extraction failed or we cannot write the file, throw exception
-        end
+      # create output response object, which is an Assembly::Image type object
+      Assembly::Image.new(output)
+    end
 
-        profile_conversion_switch = "-profile #{input_profile_file} -profile #{output_profile_file}"
+    private
+
+    def kdu_compress_default_options
+      [
+        '-precise', # forces the use of 32-bit representations
+        '-no_weights', # minimization of the MSE over all reconstructed colour components
+        '-quiet', # suppress informative messages.
+        'Creversible=no', # Disable reversible compression
+        'Cmodes=BYPASS', #
+        'Corder=RPCL', # R=resolution P=position C=component L=layer
+        'Cblk=\\{64,64\\}', # code-block dimensions; 64x64 happens to also be the default
+        'Cprecincts=\\{256,256\\},\\{256,256\\},\\{128,128\\}', # Precinct dimensions; 256x256 for the 2 highest resolution levels, defaults to 128x128 for the rest
+        'ORGgen_plt=yes', # Insert packet length information
+        '-rate 1.5', # Ratio of compressed bits to the image size
+        'Clevels=5' # Number of wavelet decomposition levels, or stages
+      ]
```

Comment:
FWIW: The Bodliens new kakadu params:

https://github.com/bodleian/image-processing/blob/master/image_processing/kakadu.py#L11-L23